src/sentry/workflow_engine/endpoints/validators/base/data_condition.py

```diff
@@ -1,30 +1,77 @@
+from sentry.workflow_engine.types import DataConditionHandler
+
+ComparisonType = TypeVar("ComparisonType")
+ConditionResult = TypeVar("ConditionResult")
+
+
+class AbstractDataConditionValidator(
+    CamelSnakeSerializer,
+    Generic[ComparisonType, ConditionResult],
+):
+    id = serializers.IntegerField(required=False)
+    type = serializers.ChoiceField(choices=[(t.value, t.value) for t in Condition])
+    comparison = serializers.JSONField(required=True)
+    condition_result = serializers.JSONField(required=True)
+    condition_group_id = serializers.IntegerField(required=True)
+
+    @abstractmethod
+    def validate_comparison(self, value: Any) -> ComparisonType:
+        pass
+
+    @abstractmethod
+    def validate_condition_result(self, value: Any) -> ConditionResult:
+        pass
+
 
+class BaseDataConditionValidator(
+    AbstractDataConditionValidator[Any, Any],
+):
+    def _get_handler(self) -> DataConditionHandler | None:
+        condition_type = self.initial_data.get("type")
+        if condition_type in CONDITION_OPS:
+            return None
 
-class BaseDataConditionValidator(CamelSnakeSerializer):
-    type = serializers.CharField(
-        required=True,
-        max_length=200,
-        help_text="Condition used to compare data value to the stored comparison value",
-    )
+        try:
+            return condition_handler_registry.get(condition_type)
+        except NoRegistrationExistsError:
+            raise serializers.ValidationError(f"Invalid condition type: {condition_type}")
 
-    @property
-    def comparison(self) -> Field:
-        raise NotImplementedError
+    def validate_comparison(self, value: Any) -> Any:
+        handler = self._get_handler()
 
-    @property
-    def result(self) -> Field:
-        raise NotImplementedError
+        if not handler:
+            raise serializers.ValidationError(
+                "Condition Operators should implement their own validators for comparison"
+            )
 
-    def validate(self, attrs):
-        attrs = super().validate(attrs)
-        return attrs
+        try:
+            return validate_json_schema(value, handler.comparison_json_schema)
+        except JsonValidationError:
+            raise serializers.ValidationError(
+                f"Value, {value} does not match JSON Schema for comparison"
+            )
 
+    def validate_condition_result(self, value: Any) -> Any:
+        handler = self._get_handler()
+        if not handler:
+            raise serializers.ValidationError(
+                "Condition Operators should implement their own validation for condition_result"
+            )
 
-class BaseDataConditionGroupValidator(CamelSnakeSerializer):
-    logic_type = serializers.CharField(required=True)
-    organization_id = serializers.IntegerField(required=True)
-    conditions = BaseDataConditionValidator(many=True)
+        try:
+            return validate_json_schema(value, handler.condition_result_schema)
```

Comment:
need to make a task to add these for each handler, should they be enforced in the pre-save signal too? 😅

Reply:
yeah - another area i'm trying to limit scope of the PR to. This was mostly to confirm it works.